docs/actionprovider.rst

```diff
@@ -1,25 +1,143 @@
+    "tasks": [
+      {
+        "function_id": "ff960aba-fa23-43d5-9cbe-3f4f91a066e1",
+        "args": [ "..." ],
+        "kwargs": { "...": "..." }
+      },
+      {
+        "function_id": "0a98fd06-edbd-11ed-abcd-0d705ebb4c49",
+        "args": [ "..." ],
+        "kwargs": { "...": "..." }
+      }
+    ],
+    "task_group_id": "97241626-8ff4-4550-9938-5909bd221869",
+    "user_endpoint_config": {
+      "min_blocks": 0,
+      "max_blocks": 1,
+      "scheduler_options": "#SBATCH --constraint=knl,quad,cache"
+    },
+    "resource_specification": {
+      "num_nodes": 2,
+      "ranks_per_node": 2,
+      "num_ranks": 4,
+      "launcher_options": "--cpu-bind quiet --mem 3072"
+    },
+    "create_queue": false
+  }
+
+The two required arguments are ``endpoint_id`` and ``tasks``. The former must be a valid
+UUID, and should point to an endpoint you have access to; the latter is a list of
+objects that represent the tasks to execute on that endpoint. Submissions must contain at
+least one task.
+
+Each task object must have a valid ``function_id``, while ``args`` and ``kwargs`` are
+optional (although if a function expects either, the flow will fail when the endpoint
+attempts to execute it). The ``args`` option is a list of arbitrary JSON data, and the
+``kwargs`` option is an object that maps argument keywords to argument values as
+arbitrary JSON data.
+
+For each task, the given endpoint will call the given function with any given arguments
+and keyword arguments, and the V3 Action Provider will return with the results of those
+calls.
+
+The other top-level arguments are optional - see the |V3SubmitRoute|_ for more
+information on what they each mean.
+
+Migrating from V2 Actions
+.........................
+
+* The V2 Action Provider accepts its ``args`` and ``kwargs`` as *strings* containing
+  JSON data, while the V3 Action Provider accepts those as raw JSON data.
+
+* With the V3 Action Provider, ``endpoint`` and ``function`` are now ``endpoint_id`` and
+  ``function_id``.
+
+* Unlike the V2 Action Provider, the V3 Action Provider only allows for submission to a
+  single endpoint at a time. If multiple endpoint submissions are needed, break those up
+  across multiple action calls.
+
+* The V3 Action Provider has no top-level ``function``, ``args`` or ``kwargs`` arguments.
+  Instead, each task, including the first, goes into the ``tasks`` array.
+
+* The V3 Action Provider does not support the ``payload`` argument. Use ``kwargs``
+  instead.
+
+Taking these points in mind, this V2 input:
+
+.. code-block:: json
+
+  {
+    "endpoint": "0c48c27a-0c2d-4028-a71b-d78697abbc4a",
+    "function": "ff960aba-fa23-43d5-9cbe-3f4f91a066e1",
+    "args": "['arg1', 'arg2']",
+    "payload": "{ 'foo': 'bar' }",
+    "tasks": [
+      {
+        "endpoint": "0c48c27a-0c2d-4028-a71b-d78697abbc4a",
+        "function": "0a98fd06-edbd-11ed-abcd-0d705ebb4c49",
+        "args": "['arg1', 'arg2']",
+        "payload": "{ 'foo': 'bar' }",
```

Comment:
```suggestion
        "payload": "{ 'foo': 'bar', 'fuzz': 'buzz' }",
```
Just to complete the example of the dict format?  If changed, also the V3 version below obviously.